helix-term/src/keymap.rs

```diff
@@ -381,30 +386,28 @@ impl Default for Keymap {
     }
 }
 
-#[derive(Debug, Clone, PartialEq, Deserialize)]
 pub struct Keymaps {
-    #[serde(flatten)]
-    pub map: HashMap<Mode, Keymap>,
-
+    pub map: Box<dyn DynAccess<HashMap<Mode, Keymap>>>,
     /// Stores pending keys waiting for the next key. This is relative to a
     /// sticky node if one is in use.
-    #[serde(skip)]
     state: Vec<KeyEvent>,
-
     /// Stores the sticky node if one is activated.
-    #[serde(skip)]
     pub sticky: Option<KeyTrieNode>,
 }
 
 impl Keymaps {
-    pub fn new(map: HashMap<Mode, Keymap>) -> Self {
+    pub fn new(map: Box<dyn DynAccess<HashMap<Mode, Keymap>>>) -> Self {
```

Comment:
I kept this as `Box<dyn DynAccess<T>>` because defining a projection function here would prevent using `arc_swap::access::Constant` which is useful for testing.